src/main/java/com/microfocus/application/automation/tools/octane/executor/TestExecutionJobCreatorService.java

```diff
@@ -351,12 +353,20 @@ public static FreeStyleProject createExecutor(DiscoveryInfo discoveryInfo) {
 			addConcurrentBuildFlag(proj);
 
 			//add build action
-			TestsToRunFramework framework = TestingToolType.UFT.equals(discoveryInfo.getTestingToolType()) ? TestsToRunFramework.MF_UFT : TestsToRunFramework.MF_MBT;
-			Builder convertedBuilder = new TestsToRunConverterBuilder(framework.value());
+			TestsToRunFramework framework = TestingToolType.UFT.equals(testingToolType) ? TestsToRunFramework.MF_UFT : TestsToRunFramework.MF_MBT;
+			Builder converterBuilder = new TestsToRunConverterBuilder(framework.value()); // uft or mbt converter
 			Builder uftRunner = new RunFromFileBuilder("${testsToRunConverted}");
-			proj.getBuildersList().add(convertedBuilder);
+			boolean isMbt = testingToolType.equals(TestingToolType.MBT);
+			Builder codelessRunner = null;
+			if (isMbt) { // in case of MBT add a second runner for codeless
+				codelessRunner = new RunFromCodelessBuilder();
+			}
+			// add steps to project
+			proj.getBuildersList().add(converterBuilder);
 			proj.getBuildersList().add(uftRunner);
-
+			if(codelessRunner != null) {
+				proj.getBuildersList().add(codelessRunner);
+			}
```

Comment:
please re-organize code, all added code can be done under "isMbt"

Reply:
donw